santa/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2018 Trail of Bits, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.9.1)
+project(santa)
+
+set(CMAKE_VERBOSE_MAKEFILE TRUE)
+
+function (main)
+  set(PROJECT_SOURCEFILES
+    src/santa.h
+    src/santa.cpp
+
+    src/extension.h
+    src/extension.cpp
+
+    src/main.cpp
+  )
+
+  set(Boost_USE_STATIC_LIBS ON)
+  set(Boost_USE_MULTITHREADED ON)
+  set(Boost_USE_STATIC_RUNTIME ON)
+  find_package(Boost REQUIRED COMPONENTS iostreams)
```

Comment:
We should document the required dependencies; do you think it is better to have it all written down in the root README or have a separate file in each extension folder?

Reply:
Concerning Boost, I'll see if we can have the full lib inside the local repository (right now we only have the following components available: filesystem, context, thread, chrono, regex, coroutine, system)